components/lock.rst

```diff
@@ -0,0 +1,306 @@
+.. index::
+   single: Lock
+   single: Components; Lock
+
+The Lock Component
+====================
+
+    The Lock Component provides a mechanism to guarantee an exclusive access
+    into a critical section. The component ships with ready to use stores for
+    the most common backends.
+
+.. versionadded:: 3.3
+    The Lock component was introduced in Symfony 3.3.
+
+Installation
+------------
+
+You can install the component in 2 different ways:
+
+* :doc:`Install it via Composer </components/using_components>` (``symfony/lock`` on `Packagist`_);
+* Use the official Git repository (https://github.com/symfony/lock).
+
+.. include:: /components/require_autoload.rst.inc
+
+Usage
+-----
+
+In order to centralize state of locks, you first need to create a ``Store``.
+Then, you can ask to this store to create a Lock for your ``resource``.
+
+The :method:`Symfony\\Component\\Lock\\LockInterface::acquire` method tries to
+acquire the lock. If the lock can not be acquired, the method throws a
+:class:`Symfony\\Component\\Lock\\Exception\\LockConflictedException`. You can
+safely call the ``acquire()`` method several times, even if you already acquired
+it.
+
+.. code-block:: php
+
+    use Symfony\Component\Lock\Store\SemaphoreStore;
+    use Symfony\Component\Lock\Exception\LockConflictedException;
+
+    $store = new SemaphoreStore();
+    $lock = $store->createLock('invoice-pdf-generation');
+
+    try {
+        $lock->acquire();
+        // the resource "invoice-pdf-generation" is locked.
+
+        // You can compute and generate invoice safely here.
+
+        $lock->release();
+    } catch (LockConflictedException $e) {
+        // the resource "invoice-pdf-generation" is already locked by another process
+    }
+
+The first argument of ``createLock`` is a string representation of the
+``resource`` to lock.
+
+.. note::
+
+    In opposition to some other implementations, the Lock Component
+    distinguishes locks instances, even when they are created from the same
+    ``resource``.
+    If you want to share a lock in several services. You have to share the
+    instance of Lock returned by the ``Store::createLock`` method.
+
+Blocking locks
+--------------
+
+You can pass an optional blocking argument as the first argument to the
+:method:`Symfony\\Component\\Lock\\LockInterface::acquire` method, which
+defaults to ``false``. If this is set to ``true``, your PHP code will wait
+infinitely until the lock is released by another process.
+
+Some ``Store`` (but not all) natively supports this feature. When they don't,
+you can decorate it with the ``RetryTillSaveStore``.
+
+.. code-block:: php
+
+    use Symfony\Component\Lock\Store\RedisStore;
+    use Symfony\Component\Lock\Store\RetryTillSaveStore;
+
+    $store = new RedisStore(new \Predis\Client('tcp://localhost:6379'));
+    $store = new RetryTillSaveStore($store);
+
+    $lock = $store->createLock('notification-flush');
+
+    $lock->acquire(true);
+
+Expirable Locks
```

Comment:
I have some doubts about the `Expirable` word. "Official" dictionaries don't include it, but Internet dictionaries include it. In any case, after reading [this discussion](http://english.stackexchange.com/questions/312081/can-expirable-be-a-word) maybe it's a good idea to rename it to `Expiring Locks`.

The key is that "expirable lock" means "it can expire" (maybe it will expire or maybe it will not) whereas "expiring lock" means "it will expire" (it's 100% sure that this lock will expire).